### PR TITLE
CUMULUS-1687 Improving async operation API

### PR DESCRIPTION
**Pull request by kkelly51**
Addresses [CUMULUS-1687: Improve asyncOps API](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1687)

## Changes

* Added "description" and "operationType" to all Async Operations (Bulk Granules, ES Reingest, Kinesis Replay)
* Added list /asyncOperations endpoint
* Added list endpoint unit test
* Updated integration tests to include "description" and "operationType"

## PR Checklist
- [ ] Unit tests
- [ ] Integration tests

### CODE REVIEW

File: packages/api/endpoints/granules.js [hunk 1]
```diff
@@ -212,11 +212,23 @@ async function bulk(req, res) {
...
   });
 
+  let description;
+
+  if (payload.query) {
+    description = `Bulk run ${payload.workflowName} on ${payload.query.size} granules`;
+  } else if (payload.ids) {
+    description = `Bulk run ${payload.workflowName} on ${payload.ids.length} granules`;
+  } else {
+    descritpion = `Bulk run on ${payload.workflowName}`;
```
Comment (laurenfrederick) [inline packages/api/endpoints/granules.js]: description is spelled incorrectly. Should have tests to cover all of these description scenarios.

File: packages/api/lambdas/index-from-database.js [hunk 2]
```diff
@@ -42,6 +42,7 @@ async function indexFromDatabase(esIndex, tables, esHost) {
   await Promise.all([
     indexModel(esClient, tables.collectionsTable, esIndex, indexer.indexCollection),
     indexModel(esClient, tables.executionsTable, esIndex, indexer.indexExecution),
+    indexModel(esClient, tables.asyncOperationsTable, esIndex, indexer.indexOperation),
```
Comment (laurenfrederick) [inline packages/api/lambdas/index-from-database.js]: This is indexing from the database into elasticsearch. I don't think we're using elasticsearch for these?

Comment (kkelly51) [reply packages/api/lambdas/index-from-database.js]: right, this was an old attempt. I'll remove it.

File: packages/api/endpoints/granules.js (hunk 1 above)
Comment (kkelly51) [reply packages/api/endpoints/granules.js]: Fixed the typo and added tests for the description

File: packages/api/endpoints/elasticsearch.js [hunk 3]
```diff
@@ -213,6 +213,8 @@ async function indexFromDatabase(req, res) {
       asyncOperationTaskDefinition: process.env.AsyncOperationTaskDefinition,
       cluster: process.env.EcsCluster,
       lambdaName: process.env.IndexFromDatabaseLambda,
+      description: 'Elasticsearch Reindex',
```
Comment (laurenfrederick) [inline packages/api/endpoints/elasticsearch.js]: I was looking through the code and this is actually an Elasticsearch index from database where we populate ES with what we have in the DB.

File: packages/api/models/schemas.js [hunk 4]
```diff
@@ -36,11 +36,13 @@ module.exports.asyncOperation = {
...
   description: 'Cumulus API AsyncOperation Table schema',
   type: 'object',
-  required: ['createdAt', 'id', 'status', 'updatedAt'],
+  required: ['createdAt', 'id', 'status', 'updatedAt', 'description', 'operationType'],
   additionalProperties: false,
   properties: {
     createdAt: { type: 'integer' },
     id: { type: 'string' },
+    description: { type: 'string' },
+    operationType: { type: 'string' },
```
Comment (laurenfrederick) [inline packages/api/models/schemas.js]: I'm wondering if this should be an enumeration so things don't deviate and then we could keep it to:
`Bulk granule`,
`ES Index`,
`Kinesis Replay`

File: packages/api/endpoints/elasticsearch.js (hunk 3 above)
Comment (kkelly51) [reply packages/api/endpoints/elasticsearch.js]: Changed

File: packages/api/models/schemas.js (hunk 4 above)
Comment (kkelly51) [reply packages/api/models/schemas.js]: changed

**Pull request merged**